src/AggregateFunctions/AggregateFunctionIf.cpp

```diff
@@ -40,28 +40,6 @@ class AggregateFunctionCombinatorIf final : public IAggregateFunctionCombinator
     }
 };
 
-/** Given an array of flags, checks if it's all zeros
```

Comment:
@Algunenano it's strange that this code was faster than `memoryIsByte` or `memoryIsZero`. The statement about `require allocating memory` is wrong, because `memcmp` does not allocate memory.

Nevertheless, even if this code is faster, it is a duplicate, and you should edit `memoryIsZero` and `memoryIsByte` instead.

Reply:
At that point I didn't know about `memoryIsZero` and `memoryIsByte`. I think I saw it a few weeks ago and thought that I should have used that instead.

I think (some time has passed so I don't remember what I was thinking) that by `require allocating memory` I meant that to do a memcmp you need something (an array full of 0's) to compare against and creating that requires an allocation (in heap or in the stack).

In any case, yes, it should use `memoryIsZero` and it performance is worse then move add the unrolled version there.